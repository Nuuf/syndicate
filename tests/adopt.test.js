/* eslint-disable @typescript-eslint/no-var-requires */
const syndicate = require('../dist');

test('changes the parent of an entity', () => {
  const root = syndicate.createRoot();
  syndicate.add(root, syndicate.create({}, 'parentA'));
  syndicate.add(root, syndicate.create({}, 'parentB'));
  syndicate.add(root, syndicate.create({}, 'child'), syndicate.find(root, 'parentA'));
  syndicate.adopt(root, syndicate.find(root, 'child'), syndicate.find(root, 'parentB'));
  expect(JSON.stringify(root, null, 2)).toMatchSnapshot();
});
