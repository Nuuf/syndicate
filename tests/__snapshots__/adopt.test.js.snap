// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the parent of an entity 1`] = `
"[
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":-1,\\\\\\"parentId\\\\\\":null,\\\\\\"index\\\\\\":0,\\\\\\"id\\\\\\":\\\\\\"root\\\\\\",\\\\\\"data\\\\\\":{\\\\\\"name\\\\\\":\\\\\\"root\\\\\\"},\\\\\\"childrenIndices\\\\\\":[1,2],\\\\\\"childrenIds\\\\\\":[\\\\\\"parentA\\\\\\",\\\\\\"parentB\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":1,\\\\\\"id\\\\\\":\\\\\\"parentA\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":0,\\\\\\"parentId\\\\\\":\\\\\\"root\\\\\\",\\\\\\"index\\\\\\":2,\\\\\\"id\\\\\\":\\\\\\"parentB\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[3],\\\\\\"childrenIds\\\\\\":[\\\\\\"child\\\\\\"]}\\",
  \\"{\\\\\\"__isSyndicateEntity__\\\\\\":true,\\\\\\"parentIndex\\\\\\":2,\\\\\\"parentId\\\\\\":\\\\\\"parentB\\\\\\",\\\\\\"index\\\\\\":3,\\\\\\"id\\\\\\":\\\\\\"child\\\\\\",\\\\\\"data\\\\\\":{},\\\\\\"childrenIndices\\\\\\":[],\\\\\\"childrenIds\\\\\\":[]}\\"
]"
`;
